ledger/ledger-consensual/src/main/kotlin/net/corda/v5/ledger/consensual/ConsensualLedgerService.kt

```diff
@@ -13,15 +14,22 @@ import net.corda.v5.ledger.consensual.transaction.ConsensualTransactionBuilder
 @DoNotImplement
 interface ConsensualLedgerService {
     /**
-     * Returns an empty [ConsensualTransactionBuilder] instance
+     * Gets a Consensual transaction builder.
+     *
+     * @return Returns a new [ConsensualTransactionBuilder] instance.
      */
     @Suspendable
     fun getTransactionBuilder(): ConsensualTransactionBuilder
 
-    /* TODO
-    fun fetchTransaction(id: SecureHash)
-    ... Vault ...
-    */
+
+    /**
+     * Fetches a transaction from the vault
+     *
+     * @param id The id of the transaction to load
+     *
+     * @return The signed transaction, if it has been recorded previously. Null if not found.
+     */
+    fun fetchTransaction(id: SecureHash) : ConsensualSignedTransaction?
```

Comment:
Is this meant to be here?

Reply:
I thought so.
Or do you see a better place for this maybe?

Reply:
No I mean, the change is about `UtxoTransactionBuilder` so this seems out of place.

Reply:
This is probably a broader scope conversation to have, but I have some concerns with regards to the naming we use. Here for example we have "fetch" and "load". In Corda 4 it was "get"

https://docs.r3.com/en/api-ref/corda/4.8/open-source/kotlin/corda/net.corda.core.node.services/-transaction-storage/get-transaction.html

I think we should try to be consistent with the naming we use, although it's hard because to my knowledge, R3 Platform Engineering hasn't defined a ubiquitous language.